rules/X++/NestedSearch.xq

```diff
@@ -0,0 +1,24 @@
+(: Finds x++ nested "while select" statements  :)
+(: @Language Xpp :)
+(: @Author bertd@microsoft.com :)
+
+<Diagnostics Category='Best practice' href='docs.microsoft.com/Socratex/NestedSearch' Version='1.0'>
+{
+  for $a in /Class | /Table | /Form | /Query
```

Comment:
Since you are doing /(Class | Table...) in the other rules, do the same here. I don't think it makes much difference one way or the other in terms of query speed (but validate that if you can).